### DISCUSSION TITLE
How does the image optimization from next/image handle file sizes?

### DISCUSSION THREAD

**Discussion by cbdeveloper** [vercel/next.js#24753] [Help] [answered]
From the docs:

- https://nextjs.org/docs/basic-features/image-optimization
- https://nextjs.org/docs/api-reference/next/image

> The Automatic Image Optimization allows for resizing, optimizing, and serving images in modern formats like WebP when the browser supports it. This **avoids shipping large images to devices with a smaller viewport**. It also allows Next.js to automatically adopt future image formats and serve them to browsers that support those formats.

My use case is the following:

- Blogging website
- Blog posts' body images
- Thumbnails

This is what I've been doing so far:

- Body images (I only upload images with max. 150kb to my storage).
- Thumbnails (I limit them on 30kb).

I currently do not have any image optimization set up.

Now that I'll be using `next/image`, does it mean that I don't need to take care of those file sizes anymore? Will it be safe to upload larger images to my storage and trust Next to do the optimization? Or should I still upload images with smaller sizes regardless of how Next.js will server them?

What if I upload a `3mb` image to my Storage? How will Next.js handle that image?

**Comment by bobarros** [comment, marked as answer]
It will do what you tell it to do.

```html
 <Image
        src="/me.png"
        alt="Picture of the author"
        width={500}
        height={500}
	quality={50}
/>
```

In most modern browsers you will get a WebP image 500x500. 
It does matter if the input is 2Mb, 3Mb, 1000w,  2000w, etc.

What you're doing, limiting file sizes like that, it's not necessary. 
It Doesn't mean that you can't ignore everything when it comes to managing images.
